src/app/games/hive/HivePieceBehavior.ts

```diff
@@ -0,0 +1,299 @@
+        for (const direction of HexaDirection.factory.all) {
+            const neighbor: Coord = coord.getNext(direction);
+            if (state.getAt(neighbor).isOccupied()) {
+                // We can jump in that direction
+                let end: Coord = neighbor;
+                while (state.getAt(end).isOccupied()) {
+                    end = end.getNext(direction);
+                }
+                moves.push(HiveMoveCoordToCoord.from(coord, end).get());
+            }
+        }
+        return moves;
+    }
+}
+
+export class HivePieceBehaviorSpider extends HivePieceBehavior {
+
+    private static INSTANCE: MGPOptional<HivePieceBehaviorSpider> = MGPOptional.empty();
+
+    public static get(): HivePieceBehaviorSpider {
+        if (this.INSTANCE.isAbsent()) {
+            this.INSTANCE = MGPOptional.of(new HivePieceBehaviorSpider());
+        }
+        return this.INSTANCE.get();
+    }
+
+    public prefixLegality(coords: Coord[], state: HiveState): MGPFallible<void> {
+        const visited: MGPSet<Coord> = new MGPSet();
+        for (let i: number = 1; i < coords.length; i++) {
+            if (state.getAt(coords[i]).isOccupied()) {
+                return MGPFallible.failure(HiveFailure.THIS_PIECE_CANNOT_CLIMB());
+            }
+            if (HexagonalUtils.areNeighbors(coords[i-1], coords[i]) === false) {
+                return MGPFallible.failure(HiveFailure.SPIDER_MUST_MOVE_ON_NEIGHBORING_SPACES());
+            }
+            if (this.haveCommonNeighbor(state, coords[i], coords[i-1]) === false) {
+                return MGPFallible.failure(HiveFailure.SPIDER_CAN_ONLY_MOVE_WITH_DIRECT_CONTACT());
+            }
+            if (this.canSlide(state, coords[i-1], coords[i]) === false) {
+                return MGPFallible.failure(HiveFailure.MUST_BE_ABLE_TO_SLIDE());
+            }
+            if (visited.contains(coords[i])) {
+                return MGPFallible.failure(HiveFailure.SPIDER_CANNOT_BACKTRACK());
+            }
+            visited.add(coords[i]);
+        }
+        return MGPFallible.success(undefined);
+    }
+
+    public moveLegality(move: HiveMoveCoordToCoord, state: HiveState): MGPFallible<void> {
+        Utils.assert(move instanceof HiveMoveSpider, 'move should be a spider move');
+        const spiderMove: HiveMoveSpider = move as HiveMoveSpider;
+        const prefixLegality: MGPFallible<void> = this.prefixLegality(spiderMove.coords, state);
+        if (prefixLegality.isFailure()) {
+            return prefixLegality;
+        }
+        return this.checkEmptyDestination(move, state);
+    }
+
+    private haveCommonNeighbor(state: HiveState, first: Coord, second: Coord): boolean {
+        const occupiedNeighbors: MGPSet<Coord> = state.getOccupiedNeighbors(first);
+        const commonNeighbor: MGPOptional<Coord> =
+            occupiedNeighbors.findAnyCommonElement(state.getOccupiedNeighbors(second));
+        return commonNeighbor.isPresent();
+    }
```

Comment:
to move in state perhaps

Reply:
aussi: on pousse le vice jusqu'au boût et on fait 

const occupiedNeighboors: MGPSet<Coord> = state.getOccupiedNeighbors(second);
const commonNeighbor: MGPOptional<Coord> = occupiedNeighbors.findAnyCommonElement(occupiedNeighboors);

Reply:
éh j'ai déjà fait ce refactor ! Et je déplace dans state, bienbu !